shell/components/form/ArrayList.vue

```diff
@@ -164,21 +164,29 @@ export default {
       { deep: true }
     );
 
+    /**
+     * Watching props.value helps to maintain internal state when slots are used
+     */
     watch(
       () => props.value,
-      (newVal) => {
+      (newVal, oldVal) => {
         lastUpdateWasFromValue.value = true;
-        const newRows = (newVal || []).map((value) => {
-          const existingRow = rows.value.find((row) => row.value === value);
 
-          if (existingRow) {
-            return { value, id: existingRow.id };
-          } else {
-            return { value, id: randomStr() };
-          }
+        // find member of the array that has been modified
+        const modifiedIdx = newVal.findIndex((current, idx) => {
+          return current !== oldVal[idx];
```

Comment:
By any chance do we need to do a deep equality check? I'm pretty sure the array can have objects instead of primitives